kuma/attachments/models.py

```diff
@@ -2,16 +2,39 @@
 from datetime import datetime
 
 from django.conf import settings
+from django.core.files.storage import default_storage
 from django.db import models
 from django.db.utils import IntegrityError
 from django.utils.encoding import python_2_unicode_compatible
 from django.utils.functional import cached_property
 from django.utils.translation import ugettext_lazy as _
 from django_mysql.models import Model as MySQLModel
+from storages.backends.s3boto3 import S3Boto3Storage
 
 from .utils import attachment_upload_to, full_attachment_url
 
 
+if settings.ATTACHMENTS_USE_S3:
+    storage = S3Boto3Storage(
+        access_key=settings.ATTACHMENTS_AWS_ACCESS_KEY_ID,
+        secret_key=settings.ATTACHMENTS_AWS_SECRET_ACCESS_KEY,
+        bucket_name=settings.ATTACHMENTS_AWS_STORAGE_BUCKET_NAME,
+        object_parameters={
+            'CacheControl': 'public, max-age=31536000, immutable',
+        },
+        auto_create_bucket=settings.ATTACHMENTS_AWS_AUTO_CREATE_BUCKET,
+        bucket_acl='public-read',  # Only used if bucket is auto-created
+        default_acl='public-read',
+        querystring_auth=False,
+        custom_domain=settings.ATTACHMENTS_AWS_S3_CUSTOM_DOMAIN,
+        secure_urls=settings.ATTACHMENTS_AWS_S3_SECURE_URLS,
+        region_name=settings.ATTACHMENTS_AWS_S3_REGION_NAME,
+        endpoint_url=settings.ATTACHMENTS_AWS_S3_ENDPOINT_URL,
+    )
+else:
+    storage = default_storage
```

Comment:
You don't need to do this. You can leave it to `None` since what you've got here is already the default (no pun intended) behavior inside `FileField`.

Reply:
So you'd prefer this?

```diff
--- a/kuma/attachments/models.py
+++ b/kuma/attachments/models.py
@@ -2,7 +2,6 @@ import os
 from datetime import datetime
 
 from django.conf import settings
-from django.core.files.storage import default_storage
 from django.db import models
 from django.db.utils import IntegrityError
 from django.utils.encoding import python_2_unicode_compatible
@@ -13,7 +12,7 @@ from storages.backends.s3boto3 import S3Boto3Storage
 
 from .utils import attachment_upload_to, full_attachment_url
 
-
+storage = None
 if settings.ATTACHMENTS_USE_S3:
     storage = S3Boto3Storage(
         access_key=settings.ATTACHMENTS_AWS_ACCESS_KEY_ID,
@@ -31,8 +30,6 @@ if settings.ATTACHMENTS_USE_S3:
         region_name=settings.ATTACHMENTS_AWS_S3_REGION_NAME,
         endpoint_url=settings.ATTACHMENTS_AWS_S3_ENDPOINT_URL,
     )
-else:
-    storage = default_storage
 
 
 @python_2_unicode_compatible
```

Reply:
FWIW, the FileField constructor looks like this:

```python
    def __init__(self, verbose_name=None, name=None, upload_to='', storage=None, **kwargs):
        self._primary_key_set_explicitly = 'primary_key' in kwargs

        self.storage = storage or default_storage
        self.upload_to = upload_to

        kwargs['max_length'] = kwargs.get('max_length', 100)
        super(FileField, self).__init__(verbose_name, name, **kwargs)
```

So falsey values get resolved to `default_storage`